pkg/reconciler/nscert/nscert_test.go

```diff
@@ -238,7 +239,8 @@ func TestReconcile(t *testing.T) {
 
 	table.Test(t, MakeFactory(func(ctx context.Context, listers *Listers, cmw configmap.Watcher) controller.Reconciler {
 		return &reconciler{
-			Base:                pkgreconciler.NewBase(ctx, controllerAgentName, cmw),
+			client:              servingclient.Get(ctx),
+			recorder:            controller.GetEventRecorder(ctx),
```

Comment:
this will be nil right? because it is not in ctx here? or is there something fancy setting a fake recorder up?

Reply:
The table test sets up a fake here

Reply:
nevermind. MakeFactory sets it up!